crates/ir/src/container.rs

```diff
@@ -0,0 +1,107 @@
+use std::mem;
+
+use crate::{
+    op_code::{IrInstruction, IrOpCode},
+    register::{chill::RegisterChill, RegisterId},
+};
+
+pub struct IrContainer {
+    _data: Vec<u8>,
+}
+
+impl IrContainer {
+    /// # Safety
+    /// This function can receive any data and it is up to the caller to ensure that the data is valid IR in valid
+    /// version.
+    pub unsafe fn from_vec(data: Vec<u8>) -> Self {
+        IrContainer { _data: data }
+    }
+
+    pub fn iter(&self) -> IrContainerIterator {
+        IrContainerIterator {
+            container: self,
+            index: 0,
+        }
+    }
+}
+
+impl<'container> IntoIterator for &'container IrContainer {
+    type Item = IrInstruction<'container>;
+    type IntoIter = IrContainerIterator<'container>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter()
+    }
+}
+
+pub struct IrContainerIterator<'container> {
+    container: &'container IrContainer,
+    index: usize,
+}
+
+impl<'container> IrContainerIterator<'container> {
```

Comment:
Lifetime names should be short and sweet so as not to pollute the code too much. I'd suggest renaming this lifetime to `'a` as it's the only lifetime in scope, or `'c` to reflect the name `IrContainer`.

Also relevant to usages below this one.